### ISSUE TITLE
multiple inputs CNN based regression problem

### ISSUE THREAD

**Issue by linbingru** [keisen/tf-keras-vis#76]
Hi,

The document says that the tf-keras-vis are now supporting the muti-inputs model (two 2D matrices). However, with the code below occurred the error. 
Thanks for your helping!!

Best,
------------------------------------------------------------------------------
# Create Gradcam object
gradcam = Gradcam(model, clone=True)

# Generate heatmap with GradCAM
cam = gradcam(Y[test], [X1[test], X2[test]], penultimate_layer=-1)

**Comment by keisen** [comment]
Hi @linbingru 

The first argument of `Gradcam#__call__()` MUST be [the score function](https://keisen.github.io/tf-keras-vis-docs/api/tf_keras_vis.utils.scores.html) as follows.


```python

function score(outputs):
    # TODO calculate score values
    return score_values

cam = gradcam(score, (X1, X2), penultimate_layer=-1)
```

Please see [the API document](https://keisen.github.io/tf-keras-vis-docs/api/tf_keras_vis.gradcam.html#tf_keras_vis.gradcam.Gradcam.__call__) and [examples](https://keisen.github.io/tf-keras-vis-docs/examples/attentions.html#GradCAM) for details.


Thanks!

**Comment by linbingru** [comment]
Hi,

Thank you very much for answering my question!
I slove the above problem, however, the new problem comes out.

"The result calculated by saliency map it can represent the weight position. However, the result calculated by gradcam was descent by value."

----------------------------------------------------------------------------------------------
from tf_keras_vis.gradcam import Gradcam

def score_function(output):
    return output[:, 0]

layer_idx='dense_3'

gradcam = Gradcam(model, clone=True)

cam = gradcam(score_function, [X2[test], X1[test]], penultimate_layer='dense_3')
----------------------------------------------------------------------------------------------

The result calculated by saliency map it can represent the weight position. However, the result calculated by gradcam was descent by value. Does the Gradcam method can also represent the saliency loacation?

----------------------------------------------------------------------------------------------
#saliency_map
print(saliency_map)
[array([[0.03965255, 0.06007694, 0.08179716, 0.08468853, 0.15592117,
         0.16889493, 0.21754322, 0.19366294, 0.2498665 , 0.26016768,
         0.36875371, 0.434514  , 0.48417194, 0.43154544, 0.54129211,
         0.50707816, 0.61562472, 0.68896445, 0.76950206, 0.82405041,
         0.84028072, 0.80198821, 0.76708105, 0.99999993, 0.76179289,
         0.8125057 , 0.84014564, 0.84142979, 0.86374242, 0.70754946,
         0.78490675, 0.85101713, 0.95127031, 0.80641341, 0.69927845,.......]],dtype=float32)]
----------------------------------------------------------------------------------------------
----------------------------------------------------------------------------------------------
#cam
print(cam)
[array([[1.00000000e+00, 9.53662038e-01, 9.07324016e-01, 8.60986114e-01,
         8.14648151e-01, 7.68310249e-01, 7.21972227e-01, 6.75634325e-01,
         6.29296362e-01, 5.82958400e-01, 5.36620438e-01, 4.90282506e-01,
         4.43944544e-01, 3.97606581e-01, 3.51268649e-01, 3.10990214e-01,
         3.01009357e-01, 2.91028500e-01, 2.81047612e-01, 2.71066755e-01,
         2.61085898e-01, 2.51105040e-01, 2.41124198e-01, 2.31143340e-01,
         2.21162468e-01, 2.11181596e-01, 2.01200753e-01, 1.91219881e-01,
         1.81239024e-01, 1.71258166e-01, 1.62492141e-01, 1.56155795e-01,
         1.49819434e-01, 1.43483087e-01, 1.37146741e-01, 1.30810395e-01,
         1.24474056e-01, 1.18137702e-01, 1.11801349e-01, 1.05465010e-01,
         9.91286561e-02, 9.27923024e-02, 8.64559487e-02, 8.01196098e-02,
         7.37832561e-02, 6.95410892e-02, 6.73930794e-02, 6.52450845e-02,
         6.30970821e-02, 6.09490909e-02, 5.88010848e-02, 5.66530824e-02,.......]],dtype=float32)]
----------------------------------------------------------------------------------------------

Best wish,
Thanks for helping me to solve this issue.

**Comment by keisen** [comment]
Hi, @linbingru .

Saliency is visualized by the gradients with respect to input values, Gradcam is visualized by the gradients with respect to the activation of the convolution layer you specified. (Because you specified `dense_3` above, Gradcam refers to the convolutional layer just before it.)
These will highlight similar positions basically, however, if the difference is big, the model might NOT be trained enough or there are some other problems.

Could you submit the result of `model.summary()` and the code snippet for reproducing the issue?

Thanks!

**Comment by linbingru** [comment]
Hi, @keisen

Thank you for your reply!  Sure, below is my model summary.

---------------------------------------------------------------------------------------------

Model: "model_13"
__________________________________________________________________________________________________
Layer (type)                    Output Shape         Param #     Connected to                     
==================================================================================================
input_1 (InputLayer)            [(None, 496, 4)]     0                                            
__________________________________________________________________________________________________
input_2 (InputLayer)            [(None, 90, 32)]     0                                            
__________________________________________________________________________________________________
conv1d_4 (Conv1D)               (None, 494, 32)      416         input_1[0][0]                    
__________________________________________________________________________________________________
conv1d (Conv1D)                 (None, 88, 32)       3104        input_2[0][0]                    
__________________________________________________________________________________________________
max_pooling1d_4 (MaxPooling1D)  (None, 247, 32)      0           conv1d_4[0][0]                   
__________________________________________________________________________________________________
max_pooling1d (MaxPooling1D)    (None, 44, 32)       0           conv1d[0][0]                     
__________________________________________________________________________________________________
conv1d_5 (Conv1D)               (None, 245, 64)      6208        max_pooling1d_4[0][0]            
__________________________________________________________________________________________________
conv1d_1 (Conv1D)               (None, 42, 64)       6208        max_pooling1d[0][0]              
__________________________________________________________________________________________________
max_pooling1d_5 (MaxPooling1D)  (None, 122, 64)      0           conv1d_5[0][0]                   
__________________________________________________________________________________________________
max_pooling1d_1 (MaxPooling1D)  (None, 21, 64)       0           conv1d_1[0][0]                   
__________________________________________________________________________________________________
conv1d_6 (Conv1D)               (None, 120, 128)     24704       max_pooling1d_5[0][0]            
__________________________________________________________________________________________________
conv1d_2 (Conv1D)               (None, 19, 128)      24704       max_pooling1d_1[0][0]            
__________________________________________________________________________________________________
max_pooling1d_6 (MaxPooling1D)  (None, 60, 128)      0           conv1d_6[0][0]                   
__________________________________________________________________________________________________
max_pooling1d_2 (MaxPooling1D)  (None, 9, 128)       0           conv1d_2[0][0]                   
__________________________________________________________________________________________________
conv1d_7 (Conv1D)               (None, 58, 64)       24640       max_pooling1d_6[0][0]            
__________________________________________________________________________________________________
conv1d_3 (Conv1D)               (None, 7, 64)        24640       max_pooling1d_2[0][0]            
__________________________________________________________________________________________________
max_pooling1d_7 (MaxPooling1D)  (None, 29, 64)       0           conv1d_7[0][0]                   
__________________________________________________________________________________________________
max_pooling1d_3 (MaxPooling1D)  (None, 3, 64)        0           conv1d_3[0][0]                   
__________________________________________________________________________________________________
dense_1 (Dense)                 (None, 29, 32)       2080        max_pooling1d_7[0][0]            
__________________________________________________________________________________________________
dense (Dense)                   (None, 3, 32)        2080        max_pooling1d_3[0][0]            
__________________________________________________________________________________________________
flatten_1 (Flatten)             (None, 928)          0           dense_1[0][0]                    
__________________________________________________________________________________________________
flatten (Flatten)               (None, 96)           0           dense[0][0]                      
__________________________________________________________________________________________________
concatenate (Concatenate)       (None, 1024)         0           flatten_1[0][0]                  
                                                                 flatten[0][0]                    
__________________________________________________________________________________________________
dense_2 (Dense)                 (None, 32)           32800       concatenate[0][0]                
__________________________________________________________________________________________________
dense_3 (Dense)                 (None, 1)            33          dense_2[0][0]                    
==================================================================================================
Total params: 151,617
Trainable params: 151,617
Non-trainable params: 0
__________________________________________________________________________________________________
None
